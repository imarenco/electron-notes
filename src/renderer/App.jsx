import React, { useEffect } from 'react';
import Notes from './components/Notes';
import { NotesProvider } from './context/NotesContext';
import styled from '@emotion/styled';

const AppContainer = styled.div`
  max-width: 960px;
  margin-right: auto;
  margin-left: auto;
  padding-right: 15px;
  padding-left: 15px;
  min-height: 100vh;
`;

const App = () => {
  return (<NotesProvider>
    <AppContainer>
      <Notes />
    </AppContainer>
  </NotesProvider>
  )
};

export default App;