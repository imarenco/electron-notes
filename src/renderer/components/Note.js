import React from 'react';
import { MdDeleteForever } from 'react-icons/md';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';

const Text = styled.p`
    word-break: break-all;
`;


const Footer = styled.div`
    position:absolute;
    bottom:0;
    right:0;
    margin: 10px;
    cursor:pointer;

`;

const Container = styled.div`
    background-color: #fef68a;
    position:relative;
    margin-bottom: 10px;
    border-radius: 10px;
    padding: 1rem;
    min-height: 170px;
    flex-direction: column;
    justify-content: space-between;
`;


const Note = ({ text, id, removeNote }) => {
    return (<Container>
        <Text data-test="text">{text}</Text>
        <Footer>
            <MdDeleteForever onClick={() => removeNote({ id: id, text: text })} size="1.3em" />
        </Footer>
    </Container>);
}

Note.propTypes = {
    text: PropTypes.string,
    id: PropTypes.string,
    removeNote: PropTypes.func,
};

export default Note;