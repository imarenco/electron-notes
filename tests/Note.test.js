
import React from 'react';
import renderer from 'react-test-renderer';
import Note from '../src/renderer/components/Note';

test('Compare Note snapshot and check properties', () => {
    const text = 'test';
    const component = renderer.create(
        <Note text={text} />,
    );

    let tree = component.toJSON();
    const comapreText = component.root.findByProps({ 'data-test': "text" }).props.children;
    expect(tree).toMatchSnapshot();
    expect(text).toEqual(comapreText);
});
