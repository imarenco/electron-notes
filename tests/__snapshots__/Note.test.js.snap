// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare Note snapshot and check properties 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-hq46nn{background-color:#fef68a;position:relative;margin-bottom:10px;border-radius:10px;padding:1rem;min-height:170px;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;justify-content:space-between;}",
      }
    }
    data-emotion="css hq46nn"
  />,
  <div
    className="css-hq46nn"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-4x1k5j{word-break:break-all;}",
        }
      }
      data-emotion="css 4x1k5j"
    />
    <p
      className="css-4x1k5j"
      data-test="text"
    >
      test
    </p>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-6x2lit{position:absolute;bottom:0;right:0;margin:10px;cursor:pointer;}",
        }
      }
      data-emotion="css 6x2lit"
    />
    <div
      className="css-6x2lit"
    >
      <svg
        fill="currentColor"
        height="1.3em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1.3em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
    </div>
  </div>,
]
`;
